config/systems-uenv/eiger-uenv.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2024 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# ReFrame CSCS settings
+#
+
+import os
+import pathlib
+import yaml
+
+import reframe.utility.osext as osext
+from reframe.core.exceptions import ConfigError
+
+uenv = os.environ.get('UENV', None)
+
+if uenv is None:
+    raise ConfigError('UENV is not set')
+
+# FIXME: Only the first image:mount pair is currenty used
+uenv_list = uenv.split(',')
+uenv_first = uenv_list[0]
+
+uenv_file, *image_mount = uenv_first.split(':')
+if len(image_mount) > 0:
+    image_mount = image_mount[0]
+else:
+    image_mount = '/user-environment'
+
+image_path = pathlib.Path(uenv_file)
+if not image_path.exists():
+    raise ConfigError(f"uenv image: '{image_path}' does not exist")
+
+image_name = image_path.stem
+
+# Options for the Slurm plugin to mount the Squashfs uenv image
+uenv_access = [f'--uenv-file={uenv_file} --uenv-mount={image_mount}']
+
+try:
+    rfm_meta = image_path.parent / f'{image_name}.yaml'
+    with open(rfm_meta) as image_envs:
+        image_environments = yaml.load(
+            image_envs.read(), Loader=yaml.BaseLoader)
+except OSError as err:
+    raise ConfigError(f"problem loading the metadata from '{rfm_meta}'")
+
+
+environs = image_environments.keys()
+environ_names =  ([f'{image_name}_{e}'for e in environs] or
+                  [f'{image_name}_builtin'])
+
+partitions = [
+    {
+        'name': 'mc',
+        'scheduler': 'squeue',
+        # 'scheduler': 'slurm',
+        'time_limit': '10m',
+        'environs': environ_names,
+        'container_platforms': [
+            {
+                'type': 'Sarus',
+            },
+            {
+                'type': 'Singularity',
+            }
+        ],
+        'max_jobs': 100,
+        'extras': {
+            'cn_memory': 500,
+        },
+        'access': [
+            '-pnormal',
+            '-Cmc',
+            f'--account={osext.osgroup()}'
+        ] + uenv_access,
```

Comment:
This doesn't need to be in the partitions anymore, we can put it in the environment. You can see an example in the description of this PR: https://github.com/reframe-hpc/reframe/pull/3152
What do you think?